gitpod-io/gitpod
# feat(gitpod-cli): report errors to ide-metrics-api

## Description
Introduce error reporting from `gp cli` (startin from `gp ports list`) to `ide-metrics-api`. Errors will be available in GCP.

## Related Issue(s)

Fixes #13707

## Follow up actions after merging
1. Add `util.logError` in all known errors 
1. Provide the `userId`
1. Find a way to capture unhandled exceptions at root level, e.g: [[1]](https://www.digitalocean.com/community/tutorials/handling-panics-in-go)


## How to test
- [Start an empty workspace in prev env](https://afalz-gitpb5e1db935e.preview.gitpod-dev.com/#https://github.com/gitpod-io/empty/) 
- in the prev env workspace, try to trigger the port exposure with `curl lama.sh | sh -s -- -p 3333`
- in the prev env workspace, break supervisor-api by replacing the following env var: `export SUPERVISOR_ADDR=localhost:22999x`
- in the prev env workspace, run `gp ports list` and make sure it does not throw any error
- [Start a workspace in prod for this PR](https://gitpod.io/#https://github.com/gitpod-io/gitpod/pull/13752)  then observe the logs for `kubectl logs ide-metrics-(...)` (you can also see the name of the ide-metrics pod via `kubectl get pods | grep ide-metrics`

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
